pawn-items.lic

```diff
@@ -0,0 +1,84 @@
+
+  def initialize
+    arg_definitions = [
+      [
+        { name: 'source', regex: /^[A-z\.\s\-']+$/i, variable: true, description: 'Source container' },
+        { name: 'noun', regex: /^[A-z\.\s\-]+$/i, optional: true, variable: true, description: 'If specified, only items with this noun will be sold.' }
+      ]
+    ]
+    args = parse_args(arg_definitions)
+
+    # Put away items in hands to mitigate accidentally selling them
+    EquipmentManager.new.return_held_gear
+    # Warn the script user that this is designed to sell items off your person
+    warn_before_sell(args.source, args.noun)
+    # Turn items into coins
+    sell_items(args.source, args.noun)
+  end
+
+  def sell_items(source, noun)
+    # If container is very full then LOOK may not list all of them.
+    # If you're moving a specific item, then sort those to the top
+    # to increase chances we find and sell all of them in one go.
+    DRC.bput("sort #{noun} in my #{source}", "are now at the top", "What were you referring to", "Please rephrase that command", "You may only sort items in your inventory") if noun
+    DRCI.get_item_list(source, 'look')
+      .map { |full_name| full_name =~ /lot of other stuff/ ? full_name : full_name.split(' ').last }
+      .select { |item| noun ? /\b#{noun}\b/ =~ item : true }
+      .each do |item|
+        # This indicates there is more items than LOOK can show
+        # and we've reached the end of what we last saw.
+        # We need to look again to see what we can now see.
+        # Keep doing this until exhaust source.
+        if item =~ /lot of other stuff/
+          sell_items(source, noun)
+          return
+        end
+        # Attempt to get the item from the source container.
+        if DRCI.get_item?(item, source)
+          unless DRCT.sell_item(Room.current.id, item)
+            # Failed. Message user and return item to source container.
+            DRC.message("Unable to sell #{item}. The merchant may not accept these items.")
+            DRCI.put_away_item?(item, source)
+            exit
+          end
+        else
+          DRC.message("Unable to get #{item} from #{source}.")
+          DRC.message("Your hands are full!") if (DRC.left_hand && DRC.right_hand)
+          exit
+        end
+      end
+  end
+
+  def warn_before_sell(source, noun)
+    # Pause script and warn that it is designed to sell items
+    # from your person to somewhere else. Item loss may occur.
+    DRC.message("WARNING: This script is designed to sell items from YOU to an NPC merchant.")
+    DRC.message("WARNING: It also has the ability to sell every item from the source container.")
+    DRC.message("WARNING: Item loss may occur. Use with caution.")
+    DRC.message("WARNING: If you want to sell loot like gems and bundles, use ;sell-loot instead.")
+    DRC.message("WARNING: If you want to transfer items between containers, use ;transfer-items or ;offload-items instead.")
+    DRC.message("\n")
+
+    unless noun
```

Comment:
I prefer to use `if` over `unless` when possible. E.g. in this case you could use `if` and swap the cases.

Reply:
Noted, thanks

Reply:
FYI @MahtraDR @asechrest

Reply:
Resolved in https://github.com/rpherbig/dr-scripts/pull/5434/commits/ed46457064b60f222f6de0079fa9ebceacbf7f2c

Reply:
Fair. Robert noted this when I created offload items. I just never came back and tweaked it.